src/hooks/useUser.ts

```diff
@@ -1,30 +1,20 @@
-import { useQuery, useQueryClient } from '@tanstack/react-query';
-import { useHttpClient } from './useHttpClient';
 import { useAuth } from './useAuth';
-import { useRestMutation } from './rest-api';
-import { User } from '../types';
+import { useRestCache, useRestMutation, useRestQuery } from './rest-api';
 
 export function useUser() {
   const { authResult } = useAuth();
-  const client = useQueryClient();
-  const { httpClient } = useHttpClient();
-
-  const result = useQuery(
-    ['user'],
-    () => httpClient.get<User>('/v1/user').then((res) => res.data),
-    {
-      enabled: !!authResult?.accessToken,
-    },
+  return useRestQuery(
+    'GET /v1/user',
+    {},
+    { enabled: !!authResult?.accessToken },
   );
+}
 
-  const { mutateAsync: updateUser } = useRestMutation('PATCH /v1/user', {
-    onSuccess: (updatedUser) => {
-      client.setQueryData(['user'], () => updatedUser);
+export const useUpdateUser = () => {
```

Comment:
I separated the mutation into a dedicated hook. Why:
- The spread on _old_ line 27 was actually killing some of the nice `react-query` typing of the query result.
- Avoiding that spread by separating this hook put the typing back into place.